internal/plugin/service/externalidentity/external_identity_data_source.go

```diff
@@ -0,0 +1,141 @@
+type externalIdentityDataSourceModel struct {
+	OrganizationID      types.String `tfsdk:"organization_id"`
+	InternalUserID      types.String `tfsdk:"internal_user_id"`
+	ExternalUserID      types.String `tfsdk:"external_user_id"`
+	ExternalServiceName types.String `tfsdk:"external_service_name"`
+}
+
+// Metadata returns the metadata for the external_identity data source.
+func (r *externalIdentityDataSource) Metadata(
+	_ context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_external_identity"
+	r.typeName = resp.TypeName
+}
+
+// TypeName returns the data source type name for the external_identity data source.
+func (r *externalIdentityDataSource) TypeName() string {
+	return r.typeName
+}
+
+// Schema defines the schema for the external_identity data source.
+func (r *externalIdentityDataSource) Schema(
+	_ context.Context,
+	_ datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Map external service user with internal aiven user",
+		Attributes: map[string]schema.Attribute{
+			"organization_id": schema.StringAttribute{
+				Description: "the organization where internal_user_id exists",
+				Required:    true,
+			},
+			"internal_user_id": schema.StringAttribute{
+				Description: "the identity of the user on aiven platform",
+				Required:    true,
+			},
+			"external_user_id": schema.StringAttribute{
+				Description: "the identity of the user on external platform",
+				Required:    true,
+			},
+			"external_service_name": schema.StringAttribute{
+				Description: "the name of the external service where external_user_id exists",
+				Required:    true,
+			},
+		},
+	}
+}
+
+// Configure sets up the external_identity data source.
+func (r *externalIdentityDataSource) Configure(
+	_ context.Context,
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*aiven.Client)
+	if !ok {
+		resp.Diagnostics = util.DiagErrorUnexpectedProviderDataType(resp.Diagnostics, req.ProviderData)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Read reads an external_identity data source.
+func (r *externalIdentityDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state externalIdentityDataSourceModel
+
+	if !util.ConfigToModel(ctx, &req.Config, &state, &resp.Diagnostics) {
+		return
+	}
+
+	organizationID := state.OrganizationID.ValueString()
+	internalUserID := state.InternalUserID.ValueString()
+	externalUserID := state.ExternalUserID.ValueString()
+	externalServiceName := state.ExternalServiceName.ValueString()
+
+	rm, err := r.client.OrganizationUser.List(ctx, organizationID)
+	if err != nil {
+		resp.Diagnostics = util.DiagErrorReadingDataSource(resp.Diagnostics, r, err)
+	}
```

Comment:
Missing early return:

```suggestion
	if err != nil {
		resp.Diagnostics = util.DiagErrorReadingDataSource(resp.Diagnostics, r, err)
		return
	}
```